### ISSUE TITLE
freeze module pass to produce parameter information

### ISSUE THREAD

**Issue by BowenBao** [pytorch/pytorch#39613]
## Description

When compared to the old `lower_graph` pass. One difference is that the `freeze_module` pass does not produce parameters information. Instead parameters are inlined in the graph as constants. 

Downstream users like ONNX depend heavily on this feature. Previously in `lower_graph` pass, the parameters are captured as graph inputs, and they are later converted to `initializers` in ONNX. Any suggestions how to achieve the same goal when switching to `freeze_module`?

## Code to repro

```python
import torch

model = torch.nn.Linear(2, 3)

s_m = torch.jit.script(model)
s_m.eval()

f_m = torch._C._freeze_module(s_m._c)
f_m.dump()

method_graph, params = torch._C._jit_pass_lower_graph(s_m.graph, s_m._c)
print(method_graph, params)

"""
freezed module signature:
...
      graph(%self : __torch__.torch.nn.modules.linear.___torch_mangle_6.Linear,
            %input.1 : Tensor):
...
lower graph signature:
...
graph(%input.1 : Tensor,
      %24 : Float(3:1),
      %25 : Float(3:2, 2:1)):
...
"""
```

cc @suo

**Comment by suo** [comment]
@bzinodev can you take a look?

**Comment by neginraoof** [comment]
Following up about this issue @bzinodev . Can you please update us about the possible solution here?

**Comment by neginraoof** [comment]
@bzinodev I want to follow up on this issue. We're looking to try the freeze_module pass for models in training mode to triage the problem. This is currently the main blocker we're facing.

**Comment by spandantiwari** [comment]
Addressed in https://github.com/pytorch/pytorch/pull/47348.
